app/src/app/present-round/present-round.component.ts

```diff
@@ -45,9 +48,69 @@ export class PresentRoundComponent implements OnInit {
         return val as boolean
       })
     )
+    this.markedDecreeAsSent = this.db.object("delegateRounds/" + this.delegateId + "/" + this.roundId + "/markedDecreeAsSent").valueChanges().pipe(
+      map(val => {
+        return val as boolean
+      })
+    )
+    this.db.list("landsraad/questions").snapshotChanges().pipe(
+      tap(
+        questionsSnapshots => {
+          this.availableQuestions = questionsSnapshots.map(qSnapshot => {
+            const question: any = qSnapshot.payload.val()
+            const qKey = qSnapshot.key
+            return {
+              ...question,
+              id: qKey
+            }
+          })
+        }
+      )
+    ).subscribe(ref => this.controlledVotingRights = this.db.list("landsraad/votingRights", ref => ref.orderByChild("controlledBy").equalTo(this.delegateId)).snapshotChanges().pipe(
+      map(
+        snapshots => {
+          return snapshots.map(snapshot => {
+            let value = snapshot.payload.val()
+            let existingQuestions = this.availableQuestions.filter(q => {
+              // check if there is existing question for this round, delegate, house 
+              // and not of the type "Bill of Particulars" (because that one is extra)
+              // we will use the same path if it exists
+              return q["byDelegateId"] === this.delegateId && q["byHouse"] === value["name"] && q["roundId"] === this.roundId && q["decretType"] !== "Bill of Particulars"
+            })
+            let path: string
```

Comment:
Tohle byla nejtezsi cast. Mit tu DB path, uz v momente kdy DecretFormComponent existuje protoze jinak apiFireForm directiva vyhazuje chybu ze path je undefined.

Tohle funguje, ale mam jeden problem v UI ze kdyz vyberou dekret, tak se to UI scrolluje trochu nahoru (protoze tim jak je tenhle kod napsany tak se ten component re-render i kdyz uz by nebylo potreba.)

Tak to bychom se na to mohli podivat jestli se to neda prepsat jinak. Ja jsem na jiny zpusob neprisla tak aby vsechny variables byly definovany v momente kdy to potrebuju.

Reply:
Tohle je blbý, to musíme přepsat. Podobný problém řeší akce, taky potřebuješ ty idčka a taky akce závisí na globálním stavu který není pro jednotlivá kola (jednotky). Vyřešilo se to tak, že v kole v admin appce máš tlačítko "Nastavit akce podle jednotek", které smaže/vytvoří akce v daném kole podle stavu jednotek. Udělal bych to podobně, tlačítko které musí admin před otevřením kola zmáčknout potom co nastaví změny v tom kdo kontroluje daný rod. Vytvoří to prázdné otázky a taky ti to výrazně zjednoduší kód tady - jenom si vyfiltruješ otázky, které jsou pro delegáta a kolo a zobrazíš to.

Reply:
Aha jasne. Takze potom uz nemusime mit vubec tu "push" option. 

Takze v kolech by mohla byt `mat-card` "dekrety" a tam neco jako "nastavit dekrety podle kontrolovanych rodu".

Ja jsem prave premyslela ze by to melo byt automaticke. A rikala jsem si ze to pujde nejak prepsat tak aby re-render neprobehl (v mobxu to jde no 😁).

Reply:
Je to všechno způsobené tím, že máme část dat per kolo a část dat globálních. Kdyby bylo všechno per-kolo, je to easy. Ale jedno tlačítko za kolo mi přijde v pohodě, už to dělám pro akce, budu to dělat i pro dekrety.

Reply:
I still don't know big part of the app, so I just assumed that these things are automatic. But it is true that making that automatic was really complex, especially with rxjs which I dont know and firebase which requires ID (btw generated ID string did not work, I tried, it had to be string from `.push`).

I will refactor this, it is easy when I don't have to generate the entry "during run time".

Reply:
Ok this is now refactored. Problem is that `setDecrees()` in the `swiss/.../round.component.ts` does not seem to be able to add questions. And it is not rights problem.

Reply:
This has been fixed too.